interactive_engine/executor/ir/core/src/glogue/pattern.rs

```diff
@@ -1199,45 +1380,53 @@ impl Pattern {
     }
 
     /// Set predicate requirement of a PatternVertex
-    pub fn set_vertex_data(&mut self, vertex_id: PatternId, opr: pb::Select) {
-        if let Some(vertex_data) = self.vertices_data.get_mut(vertex_id) {
-            vertex_data.data = opr;
+    pub fn add_vertex_params(
+        &mut self, vertex_id: PatternId, params: pb::QueryParams,
+    ) -> IrPatternResult<()> {
+        let mut vertex_data = self
+            .vertices_data
+            .remove(vertex_id)
+            .ok_or(IrPatternError::MissingPatternVertex(vertex_id))?;
+        if let Some(existed_params) = vertex_data.parameters {
+            vertex_data.parameters = Some(connect_query_params(existed_params, params));
```

Comment:
How about connect_query_params() earlier? when inserting vertex_params into `vertex_params_map`, we can connect them if the vertex already has params. And `vertex_params_map` can be simplified as a `BTreeMap<PatternId, pb::QueryParams>`

Reply:
add_vertex_params method provides the ability of connect query_params, so I think it is more simpler to use the ability of method instead of connectg query params in the translation process manually